--- origsrc/lz4-r131/programs/Makefile	2015-06-29 04:48:37.000000000 -0500
+++ src/lz4-r131/programs/Makefile	2015-10-26 12:43:40.603703700 -0500
@@ -70,7 +70,7 @@ m32: lz4c32 fullbench32 fuzzer32 framete
 
 bins: lz4 lz4c fullbench fuzzer frametest datagen
 
-all: bins m32
+all: bins # m32
 
 lz4: $(LZ4DIR)/lz4.c $(LZ4DIR)/lz4hc.c $(LZ4DIR)/lz4frame.c $(LZ4DIR)/xxhash.c bench.c lz4io.c lz4cli.c
 	$(CC)      $(FLAGS) $^ -o $@$(EXT)
--- origsrc/lz4-r131/programs/datagen.c	2015-06-29 04:48:37.000000000 -0500
+++ src/lz4-r131/programs/datagen.c	2015-10-26 12:41:51.305093000 -0500
@@ -53,7 +53,7 @@
 /**************************************
 *  OS-specific Includes
 **************************************/
-#if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(_WIN32) || defined(__CYGWIN__)
+#if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(_WIN32)
 #  include <fcntl.h>   /* _O_BINARY */
 #  include <io.h>      /* _setmode, _isatty */
 #  define SET_BINARY_MODE(file) _setmode(_fileno(file), _O_BINARY)
